src/zernike_polynomial.cpp

```diff
@@ -61,16 +75,30 @@ double ZernikePolynomials::evaluate_zernike_at_order(
         "the providede order for the zernike-polynomial is out of the stored "
         "max-order.");
   }
+
+  // get the start and end location of desired order's coefficients
+  std::size_t it_coeff_start = 0;
+  std::size_t it_coeff_end = 0;
+  if (order != 0) {
+    for (std::size_t ord = 0; ord < order; ord++) {
+      std::pair<int, int> n_and_l = get_n_and_l(ord);
+      const std::size_t n = static_cast<std::size_t>(n_and_l.first);
+      const std::size_t m = static_cast<std::size_t>(std::abs(n_and_l.second));
+      it_coeff_start += static_cast<std::size_t>( static_cast<double>(n - m) * 0.5 + 1);
+    }
+    std::pair<int, int> n_and_l = get_n_and_l(order);
+    const std::size_t n = static_cast<std::size_t>(n_and_l.first);
+    const std::size_t m = static_cast<std::size_t>(std::abs(n_and_l.second));
+    it_coeff_end = it_coeff_start + static_cast<std::size_t>(static_cast<double>(n - m) * 0.5 + 1);
+  }
+
   const std::size_t m = m_[order];
   double x_k = std::pow(x, m);
   const double x_k2 = x * x;
 
-  const std::vector<double> coeffs = Zr_coefficients_[order];
-  std::size_t k = m;
   double value = 0.;
-  for (std::size_t i = 0; i < coeffs.size(); i++) {
-    k += 2 * i;
-    value += coeffs[i] * x_k;
+  for (std::size_t i = it_coeff_start; i < it_coeff_end; i++) {
+    value += Zr_coefficients_[i] * x_k;
     x_k *= x_k2;
   }
   double mtheta = static_cast<double>(m) * theta;
```

Comment:
const ?

Reply:
I am really not sure, what exactly you refereeing to.